orx/grantlee/0.2/OrxJs.qs

```diff
@@ -0,0 +1,280 @@
+    return name;
+}
+
+function getId(name) {
+    // 'big enemy/hard-boss/boss_01' -> ['big', 'enemy', 'hard', 'boss', 'boss', '01']
+    var parts = name.split(/\/|_|-| /g);
+    
+    for (var i = 0; i < parts.length; i++) {
+        parts[i] = friendlyName(parts[i]);
+    }
+    
+    // Optimize any sequences of the same identifier.
+    // E.g. ['Boss', 'Boss', 'Boss'] -> 'Boss'
+    var clean = parts[0];
+    for (var i = 1; i < parts.length; i++) {
+        if (parts[i] !== parts[i - 1]) {
+            clean += parts[i];
+        }
+    }
+    
+    // Now we have an identifier suitable for .ini files.
+    // E.g. 'BigEnemyHardBoss01'
+    
+    return clean;
+}
+
+function section(id) {
+    return '[' + id + ']';
+}
+
+function frameSection(root, sprite, id, parentId) {
+    var frameId = id;
+    if (parentId !== undefined) {
+        frameId += '@' + parentId;
+    }
+    result = section(frameId);
+
+    if (root.exporterProperties.includeComments) {
+        result += ' ; Source: ' + sprite.fullName;
+    }
+    
+    return result;
+}
+
+function tagIf(key, value, condition) {
+    if (condition) {
+        return tag(key, value);
+    }
+}
+
+function tag(key, value) {
+    var maxTagLength = 13; // I.e. TextureOrigin
+    while (key.length < maxTagLength) {
+        key += ' ';
+    }
+    return key + ' = ' + value;
+}
+
+function pointToString(point) {
+    return '(' + point.x + ', ' + point.y + ')';
+}
+
+function sortByName(arr) {
+    return arr.slice().sort(function(a, b) {
+        return a.trimmedName.localeCompare(b.trimmedName);
+    });
+}
+
+function pivotToString(sprite) {
+    var pivot = normPointToString(sprite.pivotPointNorm);
+    if (pivot === undefined) {
+        pivot = pointToString(sprite.pivotPoint);
+    }
+    return pivot;
+}
+
+function normPointToString(point) {
+    if (point.x == 0.5 && point.y == 0.5)
+        return 'center';
+    if (point.x == 0.0 && point.y == 0.0)
+        return 'top|left';
+    if (point.x == 1.0 && point.y == 1.0)
+        return 'bottom|right';
+    if (point.x == 1.0 && point.y == 0.0)
+        return 'top|right';
+    if (point.x == 0.0 && point.y == 1.0)
+        return 'bottom|left';
+    
+    // orx wants absolute coordinates!
+}
+
+function sizeToString(size) {
+    return '(' + size.width + ', ' + size.height + ')';
+}
+
+function printAnimation(root, sprite, texture, frameCount) {
+    if (!root.settings.autodetectAnimations || frameCount < 2) {
+        return;
+    }
+    
+    var animationSetid = getContainerId(sprite.trimmedName, 'animation-set');
+    var animId = getContainerId(sprite.trimmedName, 'anim');
+    
+    append(section(animationSetid),
+           tag('Texture', texture.fullName),
+           tagIf('KeepInCache', true, root.exporterProperties.keepInCache),
+           tag('FrameSize', sizeToString(sprite.untrimmedSize)),
+           tag('KeyDuration', root.exporterProperties.keyDuration),
+           tag('Digits', digitCount(frameCount)),
```

Comment:
Are the frames output with leading `0`s? If yes, then it's all good. :) If not, `Digits` should remain at 0 (ie. not fixed-width).

Reply:
I removed this part!